sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -545,7 +545,14 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
   }
 
   private def getLocationFromStorageProps(table: CatalogTable): Option[String] = {
-    CaseInsensitiveMap(table.storage.properties).get("path")
+    val storageLoc = table.storage.locationUri.map(_.toString)
+    val storageProp = CaseInsensitiveMap(table.storage.properties).get("path")
```

Comment:
I'm a bit worried about string comparison. Are you sure the path string is always equal to the URI string? Shall we do normalization before comparing?

Reply:
We  normalization URI as  `toString => CatalogUtils.URIToString(_)` may be better ?

Reply:
to be safe, can we compare the URI? we can convert path string to URI with `CatalogUtils.stringToURI`.